### ISSUE TITLE
feedback on usage

### ISSUE THREAD

**Issue by mlaneuville** [dianna-ai/dianna#55]
I have tried using dianna on a basic CLIP pipeline (https://github.com/openai/CLIP).

CLIP takes the output of `preprocess(img)` as input, where `preprocess` is model-specific (but provided) and `img` is a PIL.Image. Once in that format, I can simply do `model.encode_image(input)`

My goal was therefore to run something like:
```
    explainer = RISE()
    heatmaps = explainer.explain_image(run_model, x)
```
Where `run_model` would include a typical CLIP call (i.e., that function can be used independently from a CLIP+dianna pipeline). What I ended up having was:
```
    explainer = RISE(preprocess_function=lambda x:np.moveaxis(x, -1, 1))
    x = np.expand_dims(np.moveaxis(images[idx].numpy(), 0, -1), axis=0).astype('double')
    heatmaps = explainer.explain_image(run_model, x, batch_size=50)
```
Where `images[idx]` would have been the usual input to CLIP.

Is that the right way to proceed in this situation? Did I miss a simpler approach?

For reference, here is my `run_model` function:
```
def run_model(x, predict=False):
    image_input = torch.tensor(np.stack(x)).cuda()

    with torch.no_grad():
        image_features = model.encode_image(image_input).float()
        image_features /= image_features.norm(dim=-1, keepdim=True)

    text_probs = (100.0 * image_features @ text_features.T).softmax(dim=-1)
    top_probs, top_labels = text_probs.topk(5, dim=-1)
    if predict:
        return top_probs, top_labels
    return text_probs.cpu().detach().numpy()
```
and `preprocess` is a function similar to
```
Compose(
    Resize(size=224, interpolation=PIL.Image.BICUBIC)
    CenterCrop(size=(224, 224))
    <function _convert_image_to_rgb at 0x14c2c470e8b0>
    ToTensor()
    Normalize(mean=(0.48145466, 0.4578275, 0.40821073), std=(0.26862954, 0.26130258, 0.27577711))
)
```

**Comment by cwmeijer** [comment]
Also see #53 and #59

**Comment by loostrum** [comment]
Hi @mlaneuville, we're thinking about implementing a way to label the axes, so that we can automatically handle moving the channels axis to the end when needed, and move it back before feeding the data to the model. Calling RISE could then be something like this:
```
x = np.random.randn(1, 3, 224, 224)
labels = {0: 'batch', 1: 'channels'}
# or alternatively provide a name for all axes as a list
# labels = ['batch ', 'channels', 'y', 'x']
explainer = RISE(axes_labels=labels)
explainer.explain_image(run_model, x, batch_size=50)
```

We'd like to hear your opinion about a feature like this.
If you want to try it out: I've implemented it for RISE on images in the `xarray` branch of `dianna`.

**Comment by mlaneuville** [comment]
I find this new approach more intuitive and the current implementation works for me, thanks!

Careful, you added 2 imports to the file which are not required I believe (lines 2-3).

**Comment by cwmeijer** [comment]
I think this can be closed. If not please reopen.
